File: problems/ocean-depth-range/solution/ocean-depth-range.py
def count_ocean_zones(depths):
    zone_counts = {
        "Sunlight Zone": 0,
        "Twilight Zone": 0,
        "Midnight Zone": 0,
        "Abyssal Zone": 0,
        "Trenches": 0
    }
    
    for depth in depths:
        # zone = classify_ocean_zone(depth)
        if depth >= 0 and depth <= 200:
            zone_counts["Sunlight Zone"] += 1
        elif depth >= 201 and depth <= 1000:
            zone_counts["Twilight Zone"] += 1
        elif depth >= 1001 and depth <= 4000:
            zone_counts["Midnight Zone"] += 1
        elif depth >= 4001 and depth <= 6000:
            zone_counts["Abyssal Zone"] += 1
        elif depth >= 6001:
            zone_counts["Trenches"] += 1
    
    return zone_counts


num_cases = int(input())
depths = []

for _ in range(num_cases):
    depths.append(int(input()))

zone_counts = count_ocean_zones(depths)

for zone, count in zone_counts.items():
    print(f"{zone}: {count}")

File: problems/ocean-depth-range/solution/OceanDepthRange.java
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Scanner;

public class OceanDepthRange {

    public static Map<String, Integer> countOceanZones(int[] depths) {
        // Use LinkedHashMap to maintain the order of the zones
        Map<String, Integer> zoneCounts = new LinkedHashMap<>();
        zoneCounts.put("Sunlight Zone", 0);
        zoneCounts.put("Twilight Zone", 0);
        zoneCounts.put("Midnight Zone", 0);
        zoneCounts.put("Abyssal Zone", 0);
        zoneCounts.put("Trenches", 0);

        // Classify each depth
        for (int depth : depths) {
            if (depth >= 0 && depth <= 200) {
                zoneCounts.put("Sunlight Zone", zoneCounts.get("Sunlight Zone") + 1);
            } else if (depth >= 201 && depth <= 1000) {
                zoneCounts.put("Twilight Zone", zoneCounts.get("Twilight Zone") + 1);
            } else if (depth >= 1001 && depth <= 4000) {
                zoneCounts.put("Midnight Zone", zoneCounts.get("Midnight Zone") + 1);
            } else if (depth >= 4001 && depth <= 6000) {
                zoneCounts.put("Abyssal Zone", zoneCounts.get("Abyssal Zone") + 1);
            } else if (depth >= 6001) {
                zoneCounts.put("Trenches", zoneCounts.get("Trenches") + 1);
            }
        }

        return zoneCounts;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read the number of depths
        int numCases = Integer.parseInt(scanner.nextLine());
        int[] depths = new int[numCases];

        // Read the depths
        for (int i = 0; i < numCases; i++) {
            depths[i] = Integer.parseInt(scanner.nextLine());
        }

        // Get the zone counts
        Map<String, Integer> zoneCounts = countOceanZones(depths);

        // Print the results in the required order
        for (Map.Entry<String, Integer> entry : zoneCounts.entrySet()) {
            System.out.println(entry.getKey() + ": " + entry.getValue());
        }

        scanner.close();
    }
}
